src/basic/path-util.c

```diff
@@ -1441,3 +1441,35 @@ int path_glob_can_match(const char *pattern, const char *prefix, char **ret) {
                 *ret = NULL;
         return false;
 }
+
+const char* default_path(void) {
+#if HAVE_SPLIT_BIN
+        bool bad = false;
+        int r;
+
+        /* Check whether /usr/sbin is separate and return the appropriate PATH.
+         * On error assume we need split usr as configured… */
+
+        FOREACH_STRING(path, "/usr/sbin", "/usr/local/sbin") {
+                _cleanup_free_ char *target = NULL;
+
+                r = readlink_malloc(path, &target);
+                if (r == -ENOENT)
+                        continue;
+                if (r >= 0 &&
+                    /* We assume that /bin is a symlink to /usr/bin, so /usr/sbin may be linked to
+                     * ../bin. The same doesn't hold for /usr/local/bin. */
+                    streq(path, "/usr/sbin") ? PATH_IN_SET(target, "bin", "./bin", "../bin", "/usr/bin")
+                                             : PATH_IN_SET(target, "bin", "./bin", "/usr/local/bin"))
```

Comment:
Hmm, why not chase()?

Reply:
It doesn't seem necessary. It really should be a trivial symlink with one of the fixed forms.

Reply:
For example, let's say that the path is "../../tmp/../bin" — what do we do then? chase() would say it's the same, but I'm not sure if we want to treat it as such. The example is a bit silly, but I don't see the benefit of making this more complicated.